src/OpenTelemetry/Metrics/MetricStreamConfiguration.cs

```diff
@@ -91,11 +95,39 @@ public class MetricStreamConfiguration
         }
     }
 
+    /// <summary>
+    /// Gets or sets a positive integer value
+    /// defining the maximum number of data points allowed to
+    /// per view.
+    /// </summary>
+    /// <remarks>
+    /// Note: If there is no matching view, but the MetricReader
+    /// defines a default cardinality limit value based on the
+    /// instrument an aggregation is created for, that value
+    /// will be used. The default value of 2000 will be used
+    /// if neither the view nor the MetricReader configures
+    /// MaxMetricPointsPerMetricStream.
+    /// </remarks>
+#if EXPOSE_EXPERIMENTAL_FEATURES
+    public
+#else
+    internal
+#endif
+    int? MaxMetricPointsPerMetricStream
```

Comment:
minor: `MetricStream` is already applied by the class name (`MetricStreamConfiguration.MaxMetricPointsPerMetricStream` is mouthful), consider one of the following names:
* `MetricStreamConfiguration.CardinalityLimit` (which is better IMHO)
* `MetricStreamConfiguration.MaxMetricPoints`
* `MetricStreamConfiguration.MaxAllowedMetricPoints`
* `MetricStreamConfiguration.MaxMetricPointsAllowed`

Reply:
While CardinalityLimit is better name, MaxMetricPoints ensures consistency across the MeterProvider setting and the View setting... but still, I am also thinking CardinalityLimit is better name and more aligned with spec.

(we added this feature much before spec, so we knew of the risk that spec will have different name than what we pick. 🤣 )

Reply:
I've updated to use CardinalityLimit to align with the spec.